### PR TITLE
rspec admin controller question

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#51119](https://edu-redmine.sun-asterisk.vn/issues/51119)

## WHAT (optional)
- Write rspec for admin controller question

## HOW
- Create file rspec for admin controller question

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Comment by ductvh-3178** [comment]
3 commit kìa a

File: db/seeds.rb [hunk 1]
```diff
@@ -17,16 +17,15 @@
...
     content: "Đây là content thứ #{n+1}",
     name: subject_name
   )
 end
 10.times do |n|
   subject_name = "Môn học của " + Faker::Name.name
-  user_id = 1
   Subject.create!(
-    user_id: User.first.id,
+    user_id: 1,
```
Comment (ghost) [inline db/seeds.rb]: (User.last || User.create....).id

File: spec/controllers/admin/questions_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    question:
+      {
+        content: "Câu hỏi số 1",
+        answers_attributes:
+        [
+          {
+            content: "Câu trả lời số 1",
+            is_correct: :true_answer
+          }
+        ]
```
Comment (ghost) [inline spec/controllers/admin/questions_controller_spec.rb]:
```suggestion
        [{content: "Câu trả lời số 1", is_correct: :true_answer}]
```

Comment (ghost) [reply spec/controllers/admin/questions_controller_spec.rb]: let :param_answers, {
{content: "Câu trả lời số 1", is_correct: :true_answer}
}

Comment (ghost) [reply spec/controllers/admin/questions_controller_spec.rb]: question:
      {
        content: "Câu hỏi số 1",
        answers_attributes:
        [param_answers]

File: spec/controllers/admin/questions_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+      end
+
+      it "should flash message failed" do
+        expect(flash[:danger]).to eq I18n.t("admin.questions.create.subject_not_found")
+      end
+    end
+
+    context "when params invalid (invalid content)" do
+      before do
+        @all_question = Question.all
```
Comment (ghost) [inline spec/controllers/admin/questions_controller_spec.rb]:
```suggestion
        @questions = Question.all
```

File: spec/controllers/admin/questions_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+    end
+
+    context "when params invalid (invalid id)" do
+      before do
+        @all_question = Question.all
+        post :create, params: invalid_id_create_params
+      end
+
+      it "should create a new question failed with invalid id" do
+        expect(@all_question.size).to eq  Question.all.size
```
Comment (ghost) [inline spec/controllers/admin/questions_controller_spec.rb]:
```suggestion
        expect(@all_question.size).to eq  Question.count
```

File: spec/controllers/admin/questions_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+      end
+
+      it "should flash message suscess" do
+        expect(flash[:info]).to eq I18n.t("admin.questions.create.create_question_successed")
+      end
+    end
+
+    context "when params invalid (invalid id)" do
+      before do
+        @all_question = Question.all
```
Comment (ghost) [inline spec/controllers/admin/questions_controller_spec.rb]:
```suggestion
        @count_questions_before_create = Question.count
```

**Pull request merged**